### PR TITLE
Ensures TIPC Kernel-Module is Deactivated

### LINKED ISSUE #579: [Feature Request] Compliance Setting: "OS must disable the Transparent Inter Process Communication kernel module"
**Is your feature request related to a problem? Please describe.**

STIG scans calling out RHEL-09-213065/OL09-00-000044/ALMA-09-030270

**Describe the solution you'd like**

Per the STIG guidance:

> To configure the system to prevent the TIPC kernel module from being loaded, run the following command:
> ```
> $ cat << EOF | tee /etc/modprobe.d/tipc.conf
> install tipc /bin/false
> blacklist tipc
> EOF
> ```

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ferricoxide**
Creates or fixes `/etc/modprobe.d/tipc.conf` as needed. Restarts `systemd-modules-load` systemd service (to re-read kernel-module config files if `/etc/modprobe.d/tipc.conf` is created or modified).

Closes #579

### CODE REVIEW

File: ash-linux/el9/STIGbyID/cat2/RHEL-09-213065.sls [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    'OEL': 'OL09-00-000044',
+    'RedHat': 'RHEL-09-213065',
+    'Rocky': 'RHEL-09-213065',
+} %}
+{%- set stig_id = stigIdByVendor[salt.grains.get('os')] %}
+{%- set helperLoc = tpldir ~ '/files' %}
+{%- set skipIt = salt.pillar.get('ash-linux:lookup:skip-stigs', []) %}
+{%- set searchDir = '/etc/modprobe.d/' %}
+{%- set modprobeFiles = [] %}
+{%- if salt.file.directory_exists('/etc/modprobe.conf') %}
```
Comment (lorengordon) [inline ash-linux/el9/STIGbyID/cat2/RHEL-09-213065.sls]: Is this actually a directory? This is different than what you were posting in slack... Not sure if this is a bad cleanup, or a new approach...?

File: ash-linux/el9/STIGbyID/cat2/RHEL-09-213065.sls [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+             kernel module.
+        ----------------------------------------
+
+{%- if stig_id in skipIt %}
+notify_{{ stig_id }}-skipSet:
+  test.show_notification:
+    - text: |
+        Handler for {{ stig_id }} has been selected for skip.
+{%- else %}
+  {%- if modprobeFiles %}
```
Comment (lorengordon) [inline ash-linux/el9/STIGbyID/cat2/RHEL-09-213065.sls]: I believe Jinja supports the for/else construct, where the else triggers if the expression is empty/false, so you don't actually need this wrapping if statement...

File: ash-linux/el9/STIGbyID/cat2/RHEL-09-213065.sls [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+             kernel module.
+        ----------------------------------------
+
+{%- if stig_id in skipIt %}
+notify_{{ stig_id }}-skipSet:
+  test.show_notification:
+    - text: |
+        Handler for {{ stig_id }} has been selected for skip.
+{%- else %}
+  {%- for modprobeFile in modprobeFiles if modprobeFiles %}
```
Comment (lorengordon) [inline ash-linux/el9/STIGbyID/cat2/RHEL-09-213065.sls]: An empty loop doesn't need to be tested if it is empty...


```suggestion
  {%- for modprobeFile in modprobeFiles %}
```
